### ISSUE TITLE
TypeError when `cache.match()` returns 304 response due to matching `If-None-Match` header

### ISSUE THREAD

**Issue by ispivey** [cloudflare/kv-asset-handler#118]
See more detail in my comment here: https://github.com/cloudflare/kv-asset-handler/pull/115/files#r445871787

tl;dr: now that we're including ETag headers on cached responses, and If-None-Match headers on requests, calls to `cache.match(request)` may return `304 Not Modified` responses which have no body. More detail about how this happens is in the [Cache API docs for the `match` method](https://developers.cloudflare.com/workers/reference/apis/cache/).

This (a) throws a `TypeError` on matches when accessing `request.body.cancel` in the current implementation, and (b) means we should probably revisit the `shouldRevalidate` logic.

You can repro by:
1. Set up a new Gatsby project
1. Follow [the docs to deploy it via Workers Sites](https://developers.cloudflare.com/workers/sites/start-from-existing/)
1. Check out the tip of master of `kv-asset-handler` in another directory
1. Run `npm link` there
1. `cd ~/my-gatsby-project && npm link @cloudflare/kv-asset-handler` to replace the npm package with your local tip of master
1. `wrangler publish` again
1. Smash reload a bunch of times and observe 500s on every 2nd request, when the request has `If-None-Match` headers and thus triggers a cache match

The simple solution is to defend against `request.body` being null in the `shouldRevalidate` logic, but I'm not sure if we need to make further updates to that logic for the sake of correctness. We'll need to address this before releasing 0.0.11.

FYI @shagamemnon @harrishancock @SukkaW

**Comment by shagamemnon** [comment]
Continued discussion here: https://github.com/cloudflare/kv-asset-handler/pull/115#issuecomment-649913302
